### PR TITLE
Add apt-get install curl in istio-proxy docker image.

### PR DESCRIPTION
**Pull request by incfly**
See #4345 for details.

### CODE REVIEW

File: pilot/docker/Dockerfile.proxy [hunk 1]
```diff
@@ -14,5 +14,9 @@ COPY envoy_bootstrap_tmpl.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
 # Allow non-root user to run envoy.
 RUN chmod 755 /usr/local/bin/envoy
 
+# We need to replacing http health-checking to a `curl localhost` command
+# when mTLS is enabled.
+RUN apt-get update && apt-get install -y curl
```
Comment (ldemailly) [inline pilot/docker/Dockerfile.proxy]: change to
```
RUN apt-get update && \
    apt-get install --no-install-recommends -y curl && \
    rm -rf /var/lib/apt/lists/*
```
like https://github.com/istio/istio/blob/master/docker/Dockerfile.xenial_debug#L8-L23
(used to be in this _debug file)

ie don't leave a huge apt/lists behind and don't install stuff not strictly needed

Comment (incfly) [reply pilot/docker/Dockerfile.proxy]: Done, thx!

**Comment by costinm** [comment]
A better way is to have a base image, like we use for the proxy_debug. We will need a way to automate the 
base image rebuild - but it doesn't have to happen on each presubmit. We have a goal of finishing the 
CirclecCI presubmit in 15 min and have fast development cycles.

**Comment by ldemailly** [comment]
we also have a goal to solve customer issues, please don’t block pr for build optimization reasons

**Comment by wattli** [comment]
@costinm @ldemailly , another solution is to use debug image when customers using http prober and mTLS is enabled.

**Comment by ldemailly** [comment]
that would be silly, debug is debug, a customer should not need to run a debug image to get a basic feature in prod

**Comment by wattli** [comment]
well, this is a corner case and the solution is short-term. If we don't want to sacrifice efficiency as Costin mentioned, I don't see better way of doing this.

**Comment by ldemailly** [comment]
how long does the apt-get takes ? does it matter compared to everything else done in the build (this is only 1 image out of many) ? we can optimize later _if needed_

**Comment by ldemailly** [comment]
@costinm please remove your block

**Comment by costinm** [comment]
I don't mind having curl - but it should be done the right way. Just have the base image build like we do for the debug base image.

**Comment by costinm** [comment]
I think this is now obsolete ? Please reopen if better option available.

**Pull request closed without merging**